src/app/prevention/strides-against/strides-against.component.ts

```diff
@@ -0,0 +1,260 @@
+              public router: Router, public dashboardService: DashboardService) {
+  }
+
+  /**
+   * Returns a random ordered Array
+   * @param  {array} array which needs to be randomized
+   */
+  randomArray(array) {
+    for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    }
+    return array;
+  }
+
+  /**
+   * Called after the man has moved linearly to the desired position
+   */
+  handleQuestion(): void {
+    let currentImage = 0;
+    let index = this.pos - 1; // Finding which object eventually landed at the first position and so on...
+    for (let i = 0; i < this.countImage; i++) {
+      if (this.paperConfig[i].imagePositionIndex === this.sortedPosition[index].index) {
+        currentImage = i;
+        break;
+      }
+    }
+    index = currentImage;
+    this.elements[index].hide(); // Hiding the objects upon Man's arrival
+    this.question = this.ObjectsConfig[index].element.question;
+    const answer = this.ObjectsConfig[index].element.answer;
+    const image = this.ObjectsConfig[index].element.messageImageURL;
+    const message = this.ObjectsConfig[index].element.messageBody;
+    const dashboardLinkMessage = this.language.messages.failure.dashboardLinkMessage;
+    const messageSuccessBody = '<img src = "' + image + '" width=auto height=200px><br><br>' + message;
+    const messageFailBody = '<img src = "' + image + '" width=auto height=200px><br><br>' + message + '' +
+      '<br><br><a href="/"> ' + dashboardLinkMessage + ' </a>';
+
+    this.openDialog().subscribe(res => {
+      if (this.result === answer) {
+        swal(this.language.messages.success.title, messageSuccessBody, 'success');
+      } else {
+        this.man.animate({transform: 'r90'}, 2000, 'linear', () => {
+          swal({
+            type: 'error',
+            title: this.language.messages.failure.title,
+            html: messageFailBody,
+            confirmButtonText: this.language.messages.failure.confirmButtonText,
+          }).then(() => location.reload());
+          this.man.toBack();
+        })
+      }
+    });
+  }
+
+  /**
+   * Opens Dialog for asking questions when called
+   */
+  openDialog(): Observable<boolean> {
+    const dialogRef = this.dialog.open(DialogComponent, {
+      data: {
+        question: this.question,
+      },
+      position: {
+        top: '300px',
+        left: '560px'
+      },
+      disableClose: true,
+      autoFocus: true,
+      width: '482px'
+    });
+    dialogRef.afterClosed().subscribe(res => this.result = res);
+    return dialogRef.afterClosed();
+  }
+
+  /**
+   * Checks for the sign of the movement of amn in X direction
+   * @param  {number} position final position of the man.
+   */
+  checkXMoveSign(position: number): number {
+    if (position <= 8 || (position <= 24 && position > 17)) {
```

Comment:
I think these comparisons can be made more efficient by having another variable called say mapForward. When he starts give it 0. As soon as he moves down make it 1, one more move down make it 0 and so on...

the +1 or -1 based on mapForward..

otherwise you could also make mapForward = -1 or +1 and just add mapForward no matter what.. that case no if required..

Reply:
@lunayach please change as per @shreyans29's suggestion

Reply:
@shreyans29 Yes! removed the function all together :+1: